contracts/src/systems/resources/contracts.cairo

```diff
@@ -269,6 +269,48 @@ mod resource_systems {
             (actual_recipient_id, hash(resources_felt_arr.span()), total_resources_weight)
         }
 
+        fn mint_if_adequate_capacity(
+            world: IWorldDispatcher, recipient_id: ID, resource: (u8, u128), check_lock: bool
+        ) -> (u128, bool) {
+            // ensure recipient is not travelling
+            get!(world, recipient_id, ArrivalTime).assert_not_travelling();
+
+            let mut success = false;
+
+            // only add to balance if receiver can carry weight
+            let (resource_type, resource_amount) = resource;
+            let mut total_resources_weight = 0;
+            total_resources_weight += WeightConfigCustomImpl::get_weight(world, resource_type, resource_amount);
+            let mut recipient_weight: Weight = get!(world, recipient_id, Weight);
+            let recipient_capacity: Capacity = get!(world, recipient_id, Capacity);
+            let recipient_quantity: Quantity = get!(world, recipient_id, Quantity);
+
+            recipient_weight.value += total_resources_weight;
```

Comment:
why do you increase the recipient weight here ?

Reply:
so that the recipient_weight in the next line `recipient_capacity.can_carry(recipient_quantity, recipient_weight)` includes the weight of the new resources. we're essentially asking, can recipient carry old weight + new weight

Reply:
ok makes sense